File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/integration/keyos/KeyOsWidevineContentProtectionIntegration.java
package com.theoplayer.contentprotectionintegration.integration.keyos;

import com.theoplayer.android.api.contentprotection.ContentProtectionIntegration;
import com.theoplayer.android.api.contentprotection.LicenseRequestCallback;
import com.theoplayer.android.api.contentprotection.Request;
import com.theoplayer.android.api.source.drm.DRMConfiguration;

public class KeyOsWidevineContentProtectionIntegration extends ContentProtectionIntegration {
    private final DRMConfiguration contentProtectionConfiguration;

    public KeyOsWidevineContentProtectionIntegration(DRMConfiguration configuration) {
        this.contentProtectionConfiguration = configuration;
    }

    @Override
    public void onLicenseRequest(Request request, LicenseRequestCallback callback) {

        if (this.contentProtectionConfiguration.getWidevine() == null) {
            throw new NullPointerException("The license acquisition URL can not be null");
        }

        request.setUrl(contentProtectionConfiguration.getWidevine().getLicenseAcquisitionURL());
        request.getHeaders().put(
                "customdata",
                contentProtectionConfiguration.getIntegrationParameters().get("customdata").toString()
        );
        callback.request(request);
    }
}

File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/integration/keyos/KeyOsWidevineContentProtectionIntegration.kt
package com.theoplayer.contentprotectionintegration.integration.keyos

import com.theoplayer.android.api.source.drm.DRMConfiguration
import com.theoplayer.android.api.contentprotection.ContentProtectionIntegration
import com.theoplayer.android.api.contentprotection.LicenseRequestCallback
import com.theoplayer.android.api.contentprotection.Request
import java.lang.NullPointerException

class KeyOsWidevineContentProtectionIntegration(private val contentProtectionConfiguration: DRMConfiguration) :
    ContentProtectionIntegration() {
    override fun onLicenseRequest(request: Request, callback: LicenseRequestCallback) {
        if (contentProtectionConfiguration.widevine == null) {
            throw NullPointerException("The license acquisition URL can not be null")
        }
        request.url = contentProtectionConfiguration.widevine!!.licenseAcquisitionURL
        request.headers["x-keyos-authorization"] =
            contentProtectionConfiguration.integrationParameters["x-keyos-authorization"].toString()
        callback.request(request)
    }
}